site.yml.sample

```diff
@@ -37,6 +37,10 @@
         name: '*'
         state: latest
 
+    # Moreover you need to use ip address of interface where etcd will be
+    # listening, to configure etcd itself.
+    - set_fact: etcd_ip_address={{hostvars[inventory_hostname]['ansible_default_ipv4']["address"]}}
```

Comment:
What will happen if ansible scripts executed from a different server?

Reply:
I agree, this looks error prone.

Reply:
@nthomas-redhat it works perfectly even from different server. Tested.

Reply:
However i will verify in another setup and update here!

Reply:
verified!

Reply:
If it works, I am good here.
@mbukatov , what you think?

Reply:
I don't think this makes sense: it defines new fact `etcd_ip_address` for all machines in ansible inventory file as an ip address of default ipv4 interface as detected by ansible. This means that value of `etcd_ip_address` fact will be different on each machine and the only host for which it could be correct (as it's not guaranteed that this default will be working, in a similar way how it's not guaranteed for the default which is there right now) is tendrl server machine.

See this example:

```
$ cat etcd-test.yml
---
- hosts: all
  user: root
  tasks:
    - set_fact: etcd_ip_address={{hostvars[inventory_hostname]['ansible_default_ipv4']["address"]}}
    - debug: msg="{{etcd_ip_address}}"

- hosts: tendrl-server
  user: root
  tasks:
    - debug: msg="{{etcd_ip_address}}"

- hosts: mbukatov-usm1-gl1.usmqe.lab.eng.brq.redhat.com
  user: root
  tasks:
    - debug: msg="{{etcd_ip_address}}"
```
```
$ ansible-playbook -i mbukatov-usm1.hosts etcd-test.yml
[DEPRECATION WARNING]: ANSIBLE_HOSTS option, The variable is misleading as it can be a list of hosts and/or paths to inventory sources . This feature will be removed in version 2.8. Deprecation warnings 
can be disabled by setting deprecation_warnings=False in ansible.cfg.

PLAY [all] ***************************************************************************************************************************************************************************************************

TASK [Gathering Facts] ***************************************************************************************************************************************************************************************
Friday 16 February 2018  17:07:45 +0100 (0:00:00.089)       0:00:00.089 ******* 
ok: [mbukatov-usm1-server.usmqe.lab.eng.brq.redhat.com]
ok: [mbukatov-usm1-client.usmqe.lab.eng.brq.redhat.com]
ok: [mbukatov-usm1-gl4.usmqe.lab.eng.brq.redhat.com]
ok: [mbukatov-usm1-gl1.usmqe.lab.eng.brq.redhat.com]
ok: [mbukatov-usm1-gl3.usmqe.lab.eng.brq.redhat.com]
ok: [mbukatov-usm1-gl2.usmqe.lab.eng.brq.redhat.com]
ok: [mbukatov-usm1-gl5.usmqe.lab.eng.brq.redhat.com]
ok: [mbukatov-usm1-gl6.usmqe.lab.eng.brq.redhat.com]

TASK [set_fact] **********************************************************************************************************************************************************************************************
Friday 16 February 2018  17:07:49 +0100 (0:00:03.979)       0:00:04.069 ******* 
ok: [mbukatov-usm1-client.usmqe.lab.eng.brq.redhat.com]
ok: [mbukatov-usm1-server.usmqe.lab.eng.brq.redhat.com]
ok: [mbukatov-usm1-gl1.usmqe.lab.eng.brq.redhat.com]
ok: [mbukatov-usm1-gl2.usmqe.lab.eng.brq.redhat.com]
ok: [mbukatov-usm1-gl3.usmqe.lab.eng.brq.redhat.com]
ok: [mbukatov-usm1-gl4.usmqe.lab.eng.brq.redhat.com]
ok: [mbukatov-usm1-gl5.usmqe.lab.eng.brq.redhat.com]
ok: [mbukatov-usm1-gl6.usmqe.lab.eng.brq.redhat.com]

TASK [debug] *************************************************************************************************************************************************************************************************
Friday 16 February 2018  17:07:49 +0100 (0:00:00.159)       0:00:04.228 ******* 
ok: [mbukatov-usm1-client.usmqe.lab.eng.brq.redhat.com] => {
    "msg": "10.37.169.25"
}
ok: [mbukatov-usm1-gl1.usmqe.lab.eng.brq.redhat.com] => {
    "msg": "10.37.169.58"
}
ok: [mbukatov-usm1-server.usmqe.lab.eng.brq.redhat.com] => {
    "msg": "10.37.169.13"
}
ok: [mbukatov-usm1-gl2.usmqe.lab.eng.brq.redhat.com] => {
    "msg": "10.37.169.60"
}
ok: [mbukatov-usm1-gl3.usmqe.lab.eng.brq.redhat.com] => {
    "msg": "10.37.169.62"
}
ok: [mbukatov-usm1-gl4.usmqe.lab.eng.brq.redhat.com] => {
    "msg": "10.37.169.64"
}
ok: [mbukatov-usm1-gl5.usmqe.lab.eng.brq.redhat.com] => {
    "msg": "10.37.169.15"
}
ok: [mbukatov-usm1-gl6.usmqe.lab.eng.brq.redhat.com] => {
    "msg": "10.37.169.35"
}

PLAY [tendrl-server] *****************************************************************************************************************************************************************************************

TASK [Gathering Facts] ***************************************************************************************************************************************************************************************
Friday 16 February 2018  17:07:49 +0100 (0:00:00.149)       0:00:04.378 ******* 
ok: [mbukatov-usm1-server.usmqe.lab.eng.brq.redhat.com]

TASK [debug] *************************************************************************************************************************************************************************************************
Friday 16 February 2018  17:07:50 +0100 (0:00:00.668)       0:00:05.047 ******* 
ok: [mbukatov-usm1-server.usmqe.lab.eng.brq.redhat.com] => {
    "msg": "10.37.169.13"
}

PLAY [mbukatov-usm1-gl1.usmqe.lab.eng.brq.redhat.com] ********************************************************************************************************************************************************

TASK [Gathering Facts] ***************************************************************************************************************************************************************************************
Friday 16 February 2018  17:07:50 +0100 (0:00:00.041)       0:00:05.088 ******* 
ok: [mbukatov-usm1-gl1.usmqe.lab.eng.brq.redhat.com]

TASK [debug] *************************************************************************************************************************************************************************************************
Friday 16 February 2018  17:07:51 +0100 (0:00:00.858)       0:00:05.947 ******* 
ok: [mbukatov-usm1-gl1.usmqe.lab.eng.brq.redhat.com] => {
    "msg": "10.37.169.58"
}

PLAY RECAP ***************************************************************************************************************************************************************************************************
mbukatov-usm1-client.usmqe.lab.eng.brq.redhat.com : ok=3    changed=0    unreachable=0    failed=0   
mbukatov-usm1-gl1.usmqe.lab.eng.brq.redhat.com : ok=5    changed=0    unreachable=0    failed=0   
mbukatov-usm1-gl2.usmqe.lab.eng.brq.redhat.com : ok=3    changed=0    unreachable=0    failed=0   
mbukatov-usm1-gl3.usmqe.lab.eng.brq.redhat.com : ok=3    changed=0    unreachable=0    failed=0   
mbukatov-usm1-gl4.usmqe.lab.eng.brq.redhat.com : ok=3    changed=0    unreachable=0    failed=0   
mbukatov-usm1-gl5.usmqe.lab.eng.brq.redhat.com : ok=3    changed=0    unreachable=0    failed=0   
mbukatov-usm1-gl6.usmqe.lab.eng.brq.redhat.com : ok=3    changed=0    unreachable=0    failed=0   
mbukatov-usm1-server.usmqe.lab.eng.brq.redhat.com : ok=5    changed=0    unreachable=0    failed=0
```